### PR TITLE
Schema doc improvements and tests

### PR DESCRIPTION
**Pull request by aarnq**
> [!warning]
> **This is a public repository, ensure not to disclose:**
>
> - [x] personal data beyond what is necessary for interacting with this pull request, nor
> - [x] business confidential information, such as customer names.

### What kind of PR is this?

**Required**: Mark one of the following that is applicable:

- [ ] kind/feature       
- [x] kind/improvement   
- [ ] kind/deprecation   
- [ ] kind/documentation 
- [ ] kind/clean-up      
- [ ] kind/bug           
- [ ] kind/other         

*Optional*: Mark one or more of the following that are applicable:

> [!important]
> Breaking changes should be marked `kind/admin-change` or `kind/dev-change` depending on type
> Critical security fixes should be marked with `kind/security`

- [ ] kind/admin-change 
- [ ] kind/dev-change   
- [ ] kind/security     
- [ ] [kind/adr]()      

### What does this PR do / why do we need this PR?

This aims to increase the quality on [the generated docs from the schemas introduced in the public docs](https://github.com/elastisys/compliantkubernetes/pull/874).

#### Information to reviewers

```bash
# to run the tests from tests/
bats unit/schema.bats
```

#### Checklist

- [x] Proper commit message prefix on all commits
  
- Change checks:
  - [x] The change is transparent
  - [ ] The change is disruptive
  - [x] The change requires no migration steps
  - [ ] The change requires migration steps
  - [ ] The change upgrades CRDs
  - [x] The change updates the config *and* the schema
- Metrics checks:
  - [ ] The metrics are still exposed and present in Grafana after the change
  - [ ] The metrics names didn't change (Grafana dashboards and Prometheus alerts are not affected)
  - [ ] The metrics names did change (Grafana dashboards and Prometheus alerts were fixed)
- Logs checks:
  - [ ] The logs do not show any errors after the change
- Pod Security Policy checks:
  - [ ] Any changed pod is covered by Pod Security Admission
  - [ ] Any changed pod is covered by Gatekeeper Pod Security Policies
  - [ ] The change does not cause any pods to be blocked by Pod Security Admission or Policies
- Network Policy checks:
  - [ ] Any changed pod is covered by Network Policies
  - [ ] The change does not cause any dropped packets in the `NetworkPolicy Dashboard`
- Audit checks:
  - [ ] The change does not cause any unnecessary Kubernetes audit events
  - [ ] The change requires changes to Kubernetes audit policy
- Falco checks:
  - [ ] The change does not cause any alerts to be generated by Falco
- Bug checks:
  - [ ] The bug fix is covered by regression tests

### CODE REVIEW

**Review by Zash — APPROVED** [review]
Looks good.

Iterating over everything and `select(has("properties"))` seems a bit fragile but probably good enough as long as we avoid calling any settings `properties` or `items` etc, **and** ensure all the object and array types have at least an empty `properties: {}` or `items: {}` since they're technically optional.

Wishlist: `pre-commit` hook enforcing `yq --prettyPrint` on the schemas, if this is in scope here

**Comment by aarnq** [comment]
> Iterating over everything and `select(has("properties"))` seems a bit fragile but probably good enough as long as we avoid calling any settings `properties` or `items` etc,...

Yeah, but it is simple enough to catch _a lot_. We could potentially avoid that if we check that it contains `type: object` for `properties` and `type: array` for `items` then we should be relatively safe.

> ...**and** ensure all the object and array types have at least an empty `properties: {}` or `items: {}` since they're technically optional.

I don't want to consider this case if they only define the type and nothing more.
If they have a different construction then we should have tests targeting it.

> Wishlist: `pre-commit` hook enforcing `yq --prettyPrint` on the schemas, if this is in scope here

No, that is not in scope of this PR.

File: config/schemas/config.yaml [hunk 1]
```diff
@@ -1198,107 +1198,277 @@ properties:
...
-          - s3
-          - gcs
-          - none
-        type: string
       swift:
+        additionalProperties: false
+        title: Swift Backend Config
+        description: |-
+          > [!note]
+          > Supported as an option only for Harbor, Rclone, and Thanos.
```
Comment (robinAwallace) [inline config/schemas/config.yaml:1281]: I do not know if not renders correctly here :thinking: 
image

Comment (aarnq) [reply config/schemas/config.yaml:1281]: Do you have the latest changes in public docs?
I added a postfix to unescape those.

Comment (robinAwallace) [reply config/schemas/config.yaml:1281]: That was why :+1:

File: config/schemas/config.yaml [hunk 2]
```diff
@@ -1,10 +1,10 @@
 $schema: https://json-schema.org/draft/2020-12/schema
 $id: https://github.com/elastisys/compliantkubernetes-apps/raw/main/config/schemas/config.yaml
-title: Compliant Kubernetes Apps settings
+title: Compliant Kubernetes Apps Config
 description: |
```
Comment (robinAwallace) [inline config/schemas/config.yaml:4]:
```suggestion
description: |-
```

File: config/schemas/config.yaml [hunk 3]
```diff
@@ -2170,653 +2657,833 @@ properties:
...
+          perf:
+            $ref: "#/properties/prometheus/$defs/diskAlerts/properties/perf"
+      s3BucketAlerts:
+        title: S3 Bucket Alerts
+        description: Configure S3 bucket alerts.
+        type: object
+        additionalProperties: false
+        properties:
+          objects:
+            $ref: "#/properties/prometheus/$defs/s3BucketAlerts/properties/object"
```
Comment (Zash) [inline config/schemas/config.yaml:3077]:
```suggestion
            $ref: "#/properties/prometheus/$defs/s3BucketAlerts/properties/objects"
```

File: config/schemas/config.yaml [hunk 4]
```diff
@@ -2888,522 +4086,880 @@ properties:
...
+
+      Shipping audit and service cluster logs requires that `objectStorage` is configured, and will use the bucket or container set in `objectStorage.buckets.audit` and `objectStorage.buckets.scLogs` respectively.
+
+      > [!note]
+      > Fluentd is installed in both service cluster and workload cluster, so this configuration applies there with some exceptions.
+    $defs:
+      logManager:
+        title: Log Manager Definitions
+        description: Definitions for log-manager compaction and retention.
+        type: objects
```
Comment (Zash) [inline config/schemas/config.yaml:4112]:
```suggestion
        type: object
```

**Comment by aarnq** [comment]
I now feel done with what I wanted to achieve in this PR, however we still have work to do.
Though before that we need to better standardise how to write and manage the schema and I'll leave that to the workshop / contrib guide issue, then we can look at refining the schema more.

File: config/schemas/config.yaml [hunk 5]
```diff
@@ -1923,916 +3125,1221 @@ properties:
...
             type: boolean
+          javaDbRegistry:
+            title: Trivy Java DB Registry
+            type: string
+          javaDbRepository:
+            title: Trivy Java DB Repository
+            type: string
           imagePullSecret:
+            title: Trivy Image Pull Secret
+            description: -|
```
Comment (Zash) [inline config/schemas/config.yaml:3185]:
```suggestion
            description: |-
```

File: config/schemas/config.yaml [hunk 6]
```diff
@@ -4148,92 +6312,129 @@ properties:
...
+            title: External DNS Ingress
+            type: boolean
+          service:
+            title: External DNS Service
+            type: boolean
+      domains:
+        title: External DNS Domains
+        description: Configure the domains External DNS should manage.
+        type: array
+        items: {}
```
Comment (robinAwallace) [inline config/schemas/config.yaml:6365]: You could add that this should be a list of strings. :slightly_smiling_face:

**Review by robinAwallace — APPROVED** [review]
LGTM :slightly_smiling_face: Good job :+1:

File: config/schemas/config.yaml [hunk 7]
```diff
@@ -1923,916 +3125,1221 @@ properties:
...
-              skipRoleSync:
-                type: boolean
+                    default: grafana_viewer
+          viewersCanEdit:
+            title: Grafana Viewers Can Edit
+            type: boolean
+            default: true
+          additionalConfigValues:
+            title: Grafana Additional Config Values
+            type: object
```
Comment (OlleLarsson) [inline config/schemas/config.yaml:3403]: Seems to be a string and not an object 
https://github.com/elastisys/compliantkubernetes-apps/blob/a2d9ee1823a7bb8c67d6697739324a71e4060061/config/config/sc-config.yaml#L158-L161

**Pull request merged**